### PR TITLE
[UI-71] Add footer feature to Combobox

### PR DESCRIPTION
**Pull request by PiotrGrzechnik**
… & Readme

#### Task

- **Issue or task referred:** UI-71
- **Feature or Bugfix:** Feature

I added possibility to add footer into Autocomplete list. User should pass node as prop.

#### Checklist

- [x] Required unit tests prepared
- [x] Documentation prepared
- [x] External licenses validated (ideally MIT)

### CODE REVIEW

File: packages/combo-box/src/Menu.js [hunk 1]
```diff
@@ -33,16 +36,18 @@ const defaultProps = {
...
  *
  * @returns {React.Element}
  */
 function Menu (props) {
-  const { buildItemId, options } = props
+  const { buildItemId, footer, options } = props
 
   // Build class name for menu component
   const clsName = buildClassName([ 'combo-box', 'menu' ])
+  const footerMenuClsName = buildClassName([ 'combo-box', 'menu--footer' ])
```
Comment (michalochman) [inline packages/combo-box/src/Menu.js]: I don't think it should be a modifier to `menu`, I'd rather create a separate class for the footer, even if it shares some styles

File: packages/combo-box/src/Menu.js [hunk 2]
```diff
@@ -57,6 +62,7 @@ function Menu (props) {
   return (
     <div className={clsName}>
       {elements}
+      {footer && <div className={footerMenuClsName}>{footer}</div>}
```
Comment (rangoo94) [inline packages/combo-box/src/Menu.js]: It should check for `footer == null` instead - what if the user would like to show, i.e. the number of selected elements? `0` will not show then, despite fact it's a correct node

File: packages/combo-box/styles/main.sass [hunk 3]
```diff
@@ -56,7 +56,13 @@
...
     color: $combo-box-menu-item-text-color
     background: $combo-box-menu-item-background-color
-    overflow: scroll
+    overflow: auto
+
+    &--footer
+      position: sticky
+      bottom: 0
+      background-color: $combo-box-background-color
+      padding: $combo-box-menu-item-padding-vertical $combo-box-menu-item-padding-horizontal
```
Comment (rangoo94) [inline packages/combo-box/styles/main.sass]: The footer should have additional variables, also this `padding` vertically is too big for regular autocomplete footer (as we discussed).

**config.sass**
```sass
$combo-box-footer-background-color: $combo-box-background-color !default
$combo-box-footer-padding-vertical: $combo-box-font-size * ($x-line-height-base - 1) / 2 !default
$combo-box-footer-padding-horizontal: $combo-box-menu-item-padding-horizontal !default
```

**main.sass**
```sass
@include module('combo-box')
  &__menu-footer
    background-color: $combo-box-footer-background-color
    padding: $combo-box-footer-padding-vertical $combo-box-footer-padding-horizontal
```

File: packages/combo-box/src/AutoComplete.js [hunk 4]
```diff
@@ -14,6 +14,9 @@ const propTypes = {
   /** Additional class name */
   className: PropTypes.string,
 
+  /** Footer of items list */
+  footer: PropTypes.node,
```
Comment (rangoo94) [inline packages/combo-box/src/AutoComplete.js]: Could you add it to `SelectBox` and `ComboBox` as well?

File: packages/combo-box/styles/main.sass [hunk 5]
```diff
@@ -56,7 +56,13 @@
     border-top: 0
     color: $combo-box-menu-item-text-color
     background: $combo-box-menu-item-background-color
-    overflow: scroll
+    overflow: auto
```
Comment (rangoo94) [inline packages/combo-box/styles/main.sass]: :+1:

File: packages/combo-box/styles/main.sass [hunk 6]
```diff
@@ -56,7 +56,13 @@
...
+
+  &__footer
+    position: sticky
```
Comment (rangoo94) [inline packages/combo-box/styles/main.sass]: `position: sticky` is not very well supported, also, in my opinion, its behavior is not really specified correctly yet

**Pull request merged**